src/generator/protocol/operations.ts

```diff
@@ -307,16 +314,17 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
   let text = `${comment(name, '// ')} creates the ${info.name} request.\n`;
   text += `func (${client}) ${name}(${generateParamsSig(sig.protocolSigs.requestMethod.params, true)}) (${sig.protocolSigs.requestMethod.returns.join(', ')}) {\n`;
   text += `\turlPath := "${op.requests![0].protocol.http!.path}"\n`;
-  if (values(aggregateParameters(op)).any((each: Parameter) => { return each.protocol.http!.in === 'path' })) {
-    // replace path parameters
+  const inPathParams = values(aggregateParameters(op)).where((each: Parameter) => { return each.protocol.http !== undefined; }).where((each: Parameter) => { return each.protocol.http!.in === 'path'; });
+  // replace path parameters
+  for (const pp of values(inPathParams)) {
+    // TODO this needs to be fixed, this is an ugly solution to guarantee that string and url imports are only added when an in path parameter is actually added
     imports.add('strings');
     imports.add('net/url');
-    for (const pp of values(aggregateParameters(op)).where((each: Parameter) => { return each.protocol.http!.in === 'path' })) {
-      text += `\turlPath = strings.ReplaceAll(urlPath, "{${pp.language.go!.serializedName}}", url.PathEscape(${formatParamValue(pp, imports)}))\n`;
-    }
+    text += `\turlPath = strings.ReplaceAll(urlPath, "{${pp.language.go!.serializedName}}", url.PathEscape(${formatParamValue(pp, imports)}))\n`;
   }
   text += `\tu.Path = path.Join(u.Path, urlPath)\n`;
-  if (values(aggregateParameters(op)).any((each: Parameter) => { return each.protocol.http!.in === 'query' })) {
+  // const inQueryParams = values(aggregateParameters(op)).where((each: Parameter) => { return each.protocol.http !== undefined && each.protocol.http!.in === 'query'; });
```

Comment:
Did this not work?  Probably better to get the list of query params once.

Reply:
Nope, this is what I reverted to the previous version. It seems to exclude query params even when they should be considered. It's a similar issue to the inPathParams